packages/angular/projects/ui-angular/src/lib/components/authenticator/components/setup-totp/setup-totp.component.ts

```diff
@@ -64,6 +69,7 @@ export class SetupTotpComponent implements OnInit {
 
       logger.info('totp code was generated:', totpCode);
       this.qrCodeSource = await QRCode.toDataURL(totpCode);
+      this.qrCodeSource$.next(this.qrCodeSource);
```

Comment:
`BehaviorSubject` is a type of observable that takes an initial value and emits current value to observers. Using it with the async pipe like`"qrCodeSource$ | async"` will automatically subscribe to new values and mark component for check when `qrCodeSource` changes, and will automatically unsubscribe when component is destroyed. It's a cleaner way of subscribing to an observable and then running change detection.